torch/csrc/jit/frontend/script_type_parser.cpp

```diff
@@ -193,9 +193,10 @@ c10::optional<std::string> ScriptTypeParser::parseBaseTypeName(
     case '.': {
       auto select = Select(expr);
       const std::string& name = select.selector().name();
-      // Special case for torch.Tensor
-      if (isTorch(select.value()) && name == "Tensor") {
-        return "Tensor";
+      // Special case for torch.Tensor and its' subclasses
+      if (isTorch(select.value()) &&
+          (name.find("Tensor") != std::string::npos)) {
```

Comment:
This isn't strict enough I think, something like `torch.xTensory` would pass this check as well. Since we are already hard coding names of these tensor classes in `string_to_type.cpp`, can we use an exhaustive list to check for here as well?

Reply:
done